daemon/daemon.go

```diff
@@ -526,10 +523,16 @@ func (d *Daemon) Stop(sigCh chan<- os.Signal) error {
 		d.snapListener.Close()
 	}
 
-	if needsFullShutdown {
-		// give time to polling clients to notice restart
+	// When shutting down the snapd listener wait until the rebootNoticeWait
+	// period has passed before snapdListener is closed to allow polling
+	// clients to access the daemon. For testing we disable this unless SNAPD_SHUTDOWN_DELAY
+	// has been set, to avoid incurring this wait for every daemon restart which happens
+	// quite often in testing.
+	if !snapdenv.Testing() || osutil.GetenvBool("SNAPD_SHUTDOWN_DELAY") {
 		time.Sleep(rebootNoticeWait)
```

Comment:
you need to mock this value much more in the tests now I think, because running the tests in the pacakge here goes from 9s to 30s

Reply:
Also if we keep snapListener closing before this instead, then we could try to account to the time it might spend waiting for hooks and substract  it from the time we wait here